mlflow/tracking/llm_utils.py

```diff
@@ -0,0 +1,93 @@
+import csv
+import logging
+import mlflow
+import os
+import shutil
+import tempfile
+
+from typing import Dict, List, Union
+
+_logger = logging.getLogger(__name__)
+
+
+def log_predictions(
+    inputs: List[Union[str, Dict[str, str]]],
+    outputs: List[str],
+    prompts: List[Union[str, Dict[str, str]]],
+) -> None:
+    """
+    Log a batch of inputs, outputs and prompts for the current evaluation run.
+    If no run is active, this method will create a new active run.
+
+    :param inputs: Union of either List of input strings or List of input dictionary
+    :param outputs: List of output strings
+    :param prompts: Union of either List of prompt strings or List of prompt dictionary
+    :returns: None
+
+    .. test-code-block:: python
+        :caption: Example
+
+        import mlflow
+
+        inputs = [
+            {
+                "question": "How do I create a Databricks cluster with UC access?",
+                "context": "Databricks clusters are ...",
+            },
+            ...,
+        ]
+
+        outputs = [
+            "<Instructions for cluster creation with UC enabled>",
+        ]
+
+        prompts = [
+            "Get Databricks documentation to answer all the questions: {input}",
+        ]
+
+
+        # Log llm predictions
+        with mlflow.start_run():
+            mlflow.llm.log_predictions(inputs, outputs, prompts)
+    """
+    if len(inputs) <= 0 or len(inputs) != len(outputs) or len(inputs) != len(prompts):
+        raise ValueError(
+            "The length of inputs, outputs and prompts must be the same and not empty."
+        )
+
+    predictions = [["inputs", "outputs", "prompts"]]
+    run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
+    LLM_ARTIFACT_NAME = "llm_predictions.csv"
+
+    try:
+        artifact_path = mlflow.artifacts.download_artifacts(
+            run_id=run_id, artifact_path=LLM_ARTIFACT_NAME
+        )
+
+        with open(artifact_path, newline="") as llm_prediction:
+            predictions = list(csv.reader(llm_prediction))
+
+        _logger.info(
+            f"Appending new inputs to already existing artifact {artifact_path} for run {run_id}."
+        )
+    except OSError:
```

Comment:
If your artifact storage is not the local file system (for example, s3), `mlflow.artifacts.download_artifacts` throws a different exception. We could use `Exception` but it can match a different exception.

Reply:
Handling this with try-catch seems challenging for now because we don't have an error like `MlflowArtifactDoesNotExist`.